id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/entity/UinDraft.java

```diff
@@ -0,0 +1,152 @@
+package io.mosip.idrepository.identity.entity;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+import javax.persistence.Basic;
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.Id;
+import javax.persistence.Lob;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+
+import org.hibernate.annotations.NotFound;
+import org.hibernate.annotations.NotFoundAction;
+import org.hibernate.annotations.Type;
+import org.springframework.data.domain.Persistable;
+
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+
+/**
+ * The Class Uin -Entity class for uin table.
+ *
+ * @author Manoj SP
+ */
+@Getter
+@Setter
+@ToString(exclude = { "biometrics", "documents" })
+@Entity
+@NoArgsConstructor
+@Table(schema = "idrepo")
+public class UinDraft implements Persistable<String>, UinInfo {
+
+	public UinDraft(String uin, String uinHash, byte[] uinData, String uinDataHash, String regId, String statusCode,
+			String createdBy, LocalDateTime createdDateTime, String updatedBy, LocalDateTime updatedDateTime, Boolean isDeleted,
+			LocalDateTime deletedDateTime, List<UinBiometricDraft> biometrics, List<UinDocumentDraft> documents) {
+		this.uin = uin;
+		this.uinHash = uinHash;
+		this.uinData = uinData.clone();
+		this.uinDataHash = uinDataHash;
+		this.regId = regId;
+		this.statusCode = statusCode;
+		this.createdBy = createdBy;
+		this.createdDateTime = createdDateTime;
+		this.updatedBy = updatedBy;
+		this.updatedDateTime = updatedDateTime;
+		this.isDeleted = isDeleted;
+		this.deletedDateTime = deletedDateTime;
+		this.biometrics = biometrics;
+		this.documents = documents;
+	}
+
+	@Id
+	@Column(insertable = false, updatable = false, nullable = false)
+	private String regId;
+
+	/** The uin. */
+	private String uin;
+
+	private String uinHash;
+
+	/** The uin data. */
+	@Lob
+	@Type(type = "org.hibernate.type.BinaryType")
+	@Basic(fetch = FetchType.LAZY)
+	@Getter(AccessLevel.NONE)
+	@Setter(AccessLevel.NONE)
+	private byte[] uinData;
+
+	/** The uin data hash. */
+	private String uinDataHash;
+
+	/** The status code. */
+	private String statusCode;
+
+	/** The created by. */
+	@Column(name = "cr_by")
+	private String createdBy;
+
+	/** The created date time. */
+	@Column(name = "cr_dtimes")
+	private LocalDateTime createdDateTime;
+
+	/** The updated by. */
+	@Column(name = "upd_by")
+	private String updatedBy;
+
+	/** The updated date time. */
+	@Column(name = "upd_dtimes")
+	private LocalDateTime updatedDateTime;
+
+	/** The is deleted. */
+	private Boolean isDeleted;
```

Comment:
Why we need this? if deleted simply move it to a history table. Also ensure to get the user based on the token

Reply:
@gsasikumar isDeleted is a column present in all tables by default. It is mostly not used but it can be used for soft delete of records.
createdBy, createdDTimes, updatedBy, updatedDTimes, isDeleted, deletedDTimes are the columns which are added to all tables by default.